src/playlist-loader.js

```diff
@@ -523,24 +488,32 @@
       time = knownEnd - time;
       for (i = endIndex; i >= 0; i--) {
         segment = this.media_.segments[i];
-        time -= segment.duration || targetDuration;
+        time -= segment.duration;
+
         if (time < 0) {
           return i;
         }
       }
-      // We haven't found a segment so load the first one
-      return 0;
+
+      // We haven't found a segment so load the first one if time is zero
+      if (time === 0) {
```

Comment:
Unfortunately no. That would select segments which we don't necessarily want because the time can equal 0 right at the end of a segment (it's the old exclusive segment end time issue) when we really want the next segment before that one.

The only time time can be 0 and we *do* want to fetch that segment is on the very first segment. It's a complicated case.

Say you have three segments:
1.  4.5 seconds long
1. 4.6 seconds long
1. 4.5 seconds long - this is the only segment you have information for
 - Ends at 13.6 exactly

The fetcher is going to try and fetch time 9.1. So we find the last segment and do `time = 13.6 - 9.1` so time is `4.5`. It then enters this loop and gets the first segment subtracting `4.5` and `time` is equal to `0` - if it fetches the last segment it enters a loop since it'll never really get the right segment.